src/Raven.Server/Storage/Schema/Updates/Server/62001/From62000.cs

```diff
@@ -0,0 +1,244 @@
+            Slice byType;
+            
+            using (StorageEnvironment.GetStaticContext(out var ctx))
+            {
+                Slice.From(ctx, "ByCreatedAt", ByteStringType.Immutable, out byCreatedAt);
+                Slice.From(ctx, "ByPostponedUntil", ByteStringType.Immutable, out byPostponedUntil);
+                Slice.From(ctx, "ByType", ByteStringType.Immutable, out byType);
+            }
+            
+            LegacyNotificationsSchema.DefineKey(new TableSchema.IndexDef
+            {
+                StartIndex = LegacyNotificationsTable.IdIndex,
+                Count = 1
+            });
+
+            LegacyNotificationsSchema.DefineIndex(new TableSchema.IndexDef // might be the same ticks, so duplicates are allowed - cannot use fixed size index
+            {
+                StartIndex = LegacyNotificationsTable.CreatedAtIndex,
+                Name = byCreatedAt
+            });
+
+            LegacyNotificationsSchema.DefineIndex(new TableSchema.IndexDef // might be the same ticks, so duplicates are allowed - cannot use fixed size index
+            {
+                StartIndex = LegacyNotificationsTable.PostponedUntilIndex,
+                Name = byPostponedUntil
+            });
+
+            NewNotificationsSchema.DefineKey(new TableSchema.IndexDef
+            {
+                StartIndex = NewNotificationsTable.IdIndex,
+                Count = 1
+            });
+
+            NewNotificationsSchema.DefineIndex(new TableSchema.IndexDef
+            {
+                StartIndex = NewNotificationsTable.CreatedAtIndex,
+                Name = byCreatedAt
+            });
+
+            NewNotificationsSchema.DefineIndex(new TableSchema.IndexDef
+            {
+                StartIndex = NewNotificationsTable.PostponedUntilIndex,
+                Name = byPostponedUntil
+            });
+            
+            NewNotificationsSchema.DefineIndex(new TableSchema.IndexDef
+            {
+                StartIndex = NewNotificationsTable.TypeIndex,
+                Name = byType
+            });
+        }
+        
+        private static string GetOldTableName(string resourceName = null)
+        {
+            return string.IsNullOrEmpty(resourceName)
+                ? OldNotificationsTableName
+                : $"{OldNotificationsTableName}.{resourceName.ToLowerInvariant()}";
+        }
+        
+        private static string GetNewTableName(string resourceName = null)
+        {
+            return string.IsNullOrEmpty(resourceName)
+                ? $"{OldNotificationsTableName}.Server"
+                : $"{OldNotificationsTableName}.Database.{resourceName.ToLowerInvariant()}";
+        }
+        
+        public bool Update(UpdateStep step)
+        {
+            var databaseNames = SchemaUpgradeExtensions.GetDatabases(step);
+
+            foreach (var databaseName in databaseNames)
+            {
+                if (ProcessDatabase(step, databaseName) == false)
```

Comment:
1. Why first returned `false` value results in breaking _entire_ loop? What about other databases? What is the result of returning `false` from `Update()`?
2. This also makes me wonder if we shouldn't do commits meanwhile after processing each database. There might be servers having huge number of databases. Since all notifications are strored in the system store it could result in potentially big transaction. 
3. To implement 2. you need a mechanism of detection whether notifications from a given database have been already migrated. To easiest will be to based this check on the presence of old table name, if it won't be available it means it got moved to new format.